stringfuzz/generator.py

```diff
@@ -43,48 +44,55 @@ def generate_node(node, language):
     if isinstance(node, ExpressionNode):
         return generate_expr(node, language)
 
-    elif isinstance(node, LiteralNode):
+    if isinstance(node, SortedVarNode):
+        return '({} {})'.format(generate_node(node.name, language), generate_node(node.sort, language))
+
+    if isinstance(node, LiteralNode):
         return generate_lit(node, language)
 
-    elif isinstance(node, IdentifierNode):
+    if isinstance(node, IdentifierNode):
+        return node.name
+
+    if isinstance(node, AtomicSortNode):
         return node.name
 
-    elif isinstance(node, SortNode):
-        return node.sort
+    if isinstance(node, CompoundSortNode):
+        return '({} {})'.format(generate_node(node.symbol, language), ' '.join(generate_node(s, language) for s in node.sorts))
 
-    elif isinstance(node, ArgsNode):
-        return '()'
+    if isinstance(node, BracketsNode):
+        return '({})'.format(' '.join(generate_node(s, language) for s in node.body))
 
-    elif isinstance(node, SettingNode):
-        return ':{}'.format(node.name)
+    if isinstance(node, SettingNode):
+        return '{}'.format(generate_node(node.name, language))
 
-    elif isinstance(node, MetaDataNode):
+    if isinstance(node, MetaDataNode):
         return node.value
 
-    elif isinstance(node, ReAllCharNode):
+    if isinstance(node, ReAllCharNode):
         if language == SMT_25_STRING:
             return 're.allchar'
         else:
             raise NotSupported(node, language)
 
+    if isinstance(node, str):
+        return node
+
     # error out on all others
-    else:
-        raise NotImplementedError('no generator for {}'.format(type(node)))
+    raise NotImplementedError('no generator for {}'.format(type(node)))
 
 def generate_lit(lit, language):
     if isinstance(lit, StringLitNode):
         return encode_string(lit.value, language)
 
-    elif isinstance(lit, BoolLitNode):
```

Comment:
why go from elif to if here?

Reply:
Mostly as a stylistic preference. I've been avoiding using `elif` after statements that return.